source/faucet/main.d

```diff
@@ -56,26 +56,90 @@ import vibe.web.rest;
 
 static immutable KeyCount = WK.Keys.byRange().length;
 
-/// Configuration parameter for Faucet
-private struct Config
+/// The keys that will be used for generating transactions
+private SecretKey[PublicKey] secret_keys;
+
+/// The configuration for faucet as a faucet and a tx generator
+private Config config;
+
+private struct TxGenerator
 {
+   /// Stats port (default: 9113)
+    ushort stats_port;
+
     /// How frequently we run our periodic task
-    static immutable interval = 30.seconds;
+    ulong send_interval;
 
     /// Between how many addresses we split a transaction by
-    static immutable count = 15;
+    uint split_count;
 
-    /// Bind address
-    public string address;
+    /// Maximum number of utxo before merging instead of splitting
+    uint merge_threshold;
 
-    /// Bind port
-    public ushort port;
-
-    /// Stats port (default: 9113)
-    public ushort stats_port = 9113;
+    /// Addresses to send the transactions to
+    string[] addresses;
 
     /// Keys from the config
-    SecretKey[PublicKey] keys;
+    string[] keys;
+
+    this (ulong send_interval, uint split_count, uint merge_threshold,
+        string[] addresses, string[] keys)
+    {
+        this.stats_port = stats_port;
+        this.send_interval = send_interval;
+        this.split_count = split_count;
+        this.merge_threshold = merge_threshold;
+        this.addresses = addresses;
+        this.keys = keys;
+    }
+
+    this (Node node) @safe
```

Comment:
is this a `yaml` Node?

Reply:
Yes, I'll update the variable name to make this clearer.

Reply:
Oh I just asked to make sure. The name change might not have been necessary because the name `node` is used in other places as well 🤔

Reply:
See the last commit :-)

Reply:
I think it was a good idea to rename as we normally have a different node in mind when we read `node`.